torchvision/transforms/functional_tensor.py

```diff
@@ -451,7 +453,11 @@ def resize(
         antialias = False
 
     if antialias and interpolation not in ["bilinear", "bicubic"]:
-        raise ValueError("Antialias option is supported for bilinear and bicubic interpolation modes only")
+        # We manually set it to False to avoid an error downstream in interpolate()
+        # This behaviour is documented: the parameter is irrelevant for modes
+        # that are not bilinear or bicubic. We used to raise an error here, but
+        # now we don't as True is the default.
```

Comment:
We used to raise an error, now we silently disable antialias. Otherwise, we'd be breaking those use-cases when we switch the default to True in 2 versions:

```py
Resize(interpolation=NEAREST)(img)  # uh oh, default changed to True, which raises an error in main branch rn.
```

This is a slight change of behaviour and it's not perfect, but I feel like the alternative is worse: we could introduce another value for `antialias` e.g. `"auto"`, which would mean "Do what PIL does i.e. apply antialias for bilinear and bicubic, and no antialias otherwise". But having that + True + False + None will bloat the API IMHO.

If it makes it any better, remember that PIL already silently switches antialiasing on or off depending on the interpolation mode (granted, they don't have an `antialias` parameter so it's not a completely fair comparison).